xonsh/built_ins.py

```diff
@@ -342,15 +345,27 @@ def _run_callable_subproc(alias, args, captured=True, prev_proc=None,
                 sys.stderr.write(rtn[1])
         return ProcProxy(rtnout, rtnerr)
 
-def is_script(fname):
-    """
-    Checks whether a file should be considered for running as a script
-    """
-    return os.path.isfile(fname) and fname != os.path.basename(fname)
 
 RE_SHEBANG = re.compile(r'#![ \t]*(.+?)$')
 
 
+def _is_runnable_name(fname):
+    return os.path.isfile(fname) and fname != os.path.basename(fname)
+
+
+def _is_binary(fname, limit=80):
+    with open(fname, 'rb') as f:
+        for i in range(limit):
+            char = f.read(1)
```

Comment:
What happens when the file is less than the limit number of bytes?  Is that what the empty string case is for?